012-OFEP-variations-and-variables.md

```diff
@@ -0,0 +1,167 @@
+## Variations and Variables
+
+## State: DRAFTING
+
+Introduce two new kinds of evaluations under individual features called **variations** and **variables**, next to feature flag's own value. This will allow OpenFeature SDKs to support a wider range of feature management use cases with already existing tools.
+
+## Background
+
+The SDK API currently exposes various type-specific methods for evaluating a feature flag's single value:
+
+```js
+import { OpenFeature } from "@openfeature/web-sdk";
+
+OpenFeature.setProvider(new YourProviderOfChoice());
+
+const client = OpenFeature.getClient();
+
+// evaluation
+const isEnabled = client.getBooleanValue("my_feature", false);
+```
+
+The evaluated value could have been of other types as well, including:
+
+- `String`
+- `Number`
+- `Object`
+
+## Proposal
+
+This proposal aims to look at feature management as a broad topic, where flags are one of the implementations.
+
+OpenFeature, in its current form, is designed to support feature flags alone with their own single value.
+
+In real world implementations of various different tools (both Open Source and SaaS), we notice that the feature flags go beyond just a single value, even if that single value can be of various different primitive types for different use cases.
+
+This is where we introduce the concept of **Variations** and **Variables**.
+
+## Kinds of evaluations
+
+### Flags
+
+This is solely for checking if the feature flag itself is enabled or not, and is always of `Boolean` type.
+
+Example:
+
+```js
+const featureKey = "my_feature";
+const defaultState = false;
+
+const isEnabled = client.isEnabled(featureKey, defaultState);
```

Comment:
if we consider features to contain a single value only, then yes it's a syntatic sugar.

but if we wish to establish three separate kinds of values, like:

- flag status (boolean)
- variation (string)
- variables (key/value pairs with different types of values)

then `.isEnabled()` ends up with a meaning of its own, which evaluates the flag's status as boolean.

Reply:
I feel like it would be very difficult to have something meaningful here cross-vendor. This feels more like the "why" that would be associated with the reason or meta-data associated with a flag evaluation. (The detailed evaluation reason would be "on").

I don't think there is a consistent idea of what "on" would be, or if all vendors would have such a concept.